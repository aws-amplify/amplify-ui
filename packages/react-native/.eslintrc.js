module.exports = {
  ignorePatterns: ['dist', '.eslintrc.js', '*.config.js', 'jest.setup.js'],
  extends: ['@aws-amplify/amplify-ui/react'],
  overrides: [
    {
      extends: ['@aws-amplify/amplify-ui/jest'],
      files: ['**/__mocks__/**', '**/__tests__/**'],
    },
  ],
  // point to local tsconfig
  parserOptions: { project: ['tsconfig.json'], tsconfigRootDir: __dirname },
};
