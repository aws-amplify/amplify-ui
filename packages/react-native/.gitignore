dist
.watchmanconfig

# ignore jest generated coverage directory
coverage 
