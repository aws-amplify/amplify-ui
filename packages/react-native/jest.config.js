module.exports = {
  preset: 'react-native',
  modulePathIgnorePatterns: ['<rootDir>/dist/'],
  collectCoverage: true,
  collectCoverageFrom: [
    '<rootDir>/src/**/*.{js,jsx,ts,tsx}',
    '!<rootDir>/src/**/*{c,C}onstants.ts',
  ],
  moduleNameMapper: {
    '^react$': '<rootDir>/node_modules/react',
    '^react-native$': '<rootDir>/node_modules/react-native',
  },
  coverageThreshold: {
    global: {
      branches: 90,
      functions: 90,
      lines: 90,
      statements: 90,
    },
  },
  setupFiles: ['<rootDir>/jest.setup.js'],
};
