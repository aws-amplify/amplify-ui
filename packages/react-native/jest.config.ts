import { Config } from 'jest';

const config: Config = {
  preset: 'react-native',
  modulePathIgnorePatterns: ['<rootDir>/dist/'],
  collectCoverage: true,
  collectCoverageFrom: [
    '<rootDir>/src/**/*.{js,jsx,ts,tsx}',
    '!<rootDir>/src/**/*{c,C}onstants.ts',
  ],
  moduleNameMapper: {
    '^react-native$': '<rootDir>/node_modules/react-native',
    '^uuid$': '<rootDir>/../../node_modules/uuid',
  },
  modulePaths: ['<rootDir>/node_modules/'],
  coverageThreshold: {
    global: {
      branches: 90,
      functions: 90,
      lines: 90,
      statements: 90,
    },
  },
  setupFiles: ['<rootDir>/jest.setup.ts'],
};

export default config;
