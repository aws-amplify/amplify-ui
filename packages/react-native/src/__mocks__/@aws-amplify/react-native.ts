export default {
  loadUrlPolyfill: jest.fn(),
};
