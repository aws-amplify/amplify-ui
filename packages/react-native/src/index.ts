export {
  Authenticator,
  AuthenticatorProps,
  useAuthenticator,
  UseAuthenticator,
  withAuthenticator,
  WithAuthenticatorOptions,
} from './Authenticator';
export * from './InAppMessaging';
export {
  Theme,
  ThemeProvider,
  defaultDarkModeOverride,
  defaultTheme,
  useTheme,
} from './theme';
