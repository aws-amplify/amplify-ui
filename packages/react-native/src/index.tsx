import React from 'react';
import { Text, View } from 'react-native';

export function HelloWorld(): JSX.Element {
  return (
    <View>
      <Text>Hello World!</Text>
    </View>
  );
}
