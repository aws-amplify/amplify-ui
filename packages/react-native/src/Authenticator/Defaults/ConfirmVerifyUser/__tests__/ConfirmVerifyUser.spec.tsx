import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import { authenticatorTextUtil } from '@aws-amplify/ui';
import { ConfirmVerifyUser } from '..';

const {
  getAccountRecoveryInfoText,
  getSkipText,
  getSubmitText,
  getSubmittingText,
} = authenticatorTextUtil;

const props = {
  fields: [],
  FormFields: ConfirmVerifyUser.FormFields,
  Footer: ConfirmVerifyUser.Footer,
  handleBlur: jest.fn(),
  handleChange: jest.fn(),
  handleSubmit: jest.fn(),
  hasValidationErrors: false,
  Header: ConfirmVerifyUser.Header,
  isPending: false,
  skipVerification: jest.fn(),
};

describe('ConfirmVerifyUser', () => {
  it('renders as expected', () => {
    const { toJSON, getByRole, getByText } = render(
      <ConfirmVerifyUser {...props} />
    );
    expect(toJSON()).toMatchSnapshot();

    expect(getByRole('header')).toBeDefined();
    expect(getByText(getAccountRecoveryInfoText())).toBeDefined();
    expect(getByText(getSubmitText())).toBeDefined();
    expect(getByText(getSkipText())).toBeDefined();
  });

  it('renders as expected with errors', () => {
    const error = 'Something went wrong';
    const { toJSON, getByRole, getByText } = render(
      <ConfirmVerifyUser {...props} error={error} />
    );
    expect(toJSON()).toMatchSnapshot();

    expect(getByRole('alert')).toBeDefined();
    expect(getByText(error)).toBeDefined();
  });

  it('handles skip verification button', () => {
    const skipVerificationMock = jest.fn();

    const { getByText } = render(
      <ConfirmVerifyUser {...props} skipVerification={skipVerificationMock} />
    );

    const button = getByText(getSkipText());
    fireEvent.press(button);
    expect(skipVerificationMock).toBeCalledTimes(1);
  });

  it('renders as expected when isPending is true', () => {
    const { toJSON, getByText } = render(
      <ConfirmVerifyUser {...props} isPending />
    );
    expect(toJSON()).toMatchSnapshot();

    expect(getByText(getSubmittingText())).toBeDefined();
  });
});
