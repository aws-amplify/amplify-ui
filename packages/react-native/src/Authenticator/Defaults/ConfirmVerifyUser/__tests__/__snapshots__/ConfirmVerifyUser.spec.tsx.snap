// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmVerifyUser renders as expected 1`] = `
Array [
  <Text
    accessibilityRole="header"
    style={
      Array [
        Object {
          "color": "hsl(210, 50%, 10%)",
        },
        Object {
          "fontSize": 24,
          "fontWeight": "300",
          "lineHeight": 36,
        },
        Object {
          "marginVertical": 12,
          "paddingHorizontal": 12,
        },
      ]
    }
  >
    Account recovery requires verified contact information
  </Text>,
  <View
    style={
      Object {
        "paddingBottom": 8,
      }
    }
  />,
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "hsl(190, 95%, 30%)",
          "borderRadius": 4,
          "justifyContent": "center",
          "paddingHorizontal": 16,
          "paddingVertical": 12,
        },
        undefined,
        Object {
          "marginHorizontal": 16,
          "marginVertical": 12,
        },
      ]
    }
    testID="amplify__button"
  >
    <Text
      style={
        Array [
          Object {
            "color": "hsl(0, 0%, 100%)",
            "fontSize": 16,
            "fontWeight": "700",
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Submit
    </Text>
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 4,
            "justifyContent": "center",
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          },
          undefined,
          Object {
            "marginVertical": 8,
            "minWidth": "50%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "hsl(190, 95%, 30%)",
              "fontSize": 16,
              "fontWeight": "700",
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Skip
      </Text>
    </View>
  </View>,
]
`;

exports[`ConfirmVerifyUser renders as expected when isPending is true 1`] = `
Array [
  <Text
    accessibilityRole="header"
    style={
      Array [
        Object {
          "color": "hsl(210, 50%, 10%)",
        },
        Object {
          "fontSize": 24,
          "fontWeight": "300",
          "lineHeight": 36,
        },
        Object {
          "marginVertical": 12,
          "paddingHorizontal": 12,
        },
      ]
    }
  >
    Account recovery requires verified contact information
  </Text>,
  <View
    style={
      Object {
        "paddingBottom": 8,
      }
    }
  />,
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "hsl(190, 95%, 30%)",
          "borderRadius": 4,
          "justifyContent": "center",
          "paddingHorizontal": 16,
          "paddingVertical": 12,
        },
        undefined,
        Object {
          "marginHorizontal": 16,
          "marginVertical": 12,
        },
      ]
    }
    testID="amplify__button"
  >
    <Text
      style={
        Array [
          Object {
            "color": "hsl(0, 0%, 100%)",
            "fontSize": 16,
            "fontWeight": "700",
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Submitting
    </Text>
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 4,
            "justifyContent": "center",
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          },
          undefined,
          Object {
            "marginVertical": 8,
            "minWidth": "50%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "hsl(190, 95%, 30%)",
              "fontSize": 16,
              "fontWeight": "700",
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Skip
      </Text>
    </View>
  </View>,
]
`;

exports[`ConfirmVerifyUser renders as expected with errors 1`] = `
Array [
  <Text
    accessibilityRole="header"
    style={
      Array [
        Object {
          "color": "hsl(210, 50%, 10%)",
        },
        Object {
          "fontSize": 24,
          "fontWeight": "300",
          "lineHeight": 36,
        },
        Object {
          "marginVertical": 12,
          "paddingHorizontal": 12,
        },
      ]
    }
  >
    Account recovery requires verified contact information
  </Text>,
  <View
    style={
      Object {
        "paddingBottom": 8,
      }
    }
  />,
  <View
    accessibilityRole="alert"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "hsl(0, 75%, 85%)",
          "flexDirection": "row",
          "paddingHorizontal": 8,
          "paddingVertical": 12,
        },
        Object {
          "marginHorizontal": 12,
        },
      ]
    }
    testID="amplify__error-message"
  >
    <Image
      accessibilityRole="image"
      source={
        Object {
          "testUri": "../../../src/assets/icons/error.png",
        }
      }
      style={
        Array [
          Object {
            "height": 20,
            "resizeMode": "contain",
            "tintColor": undefined,
            "width": 20,
          },
          Array [
            Object {
              "margin": 8,
              "tintColor": "hsl(0, 100%, 20%)",
            },
            Object {},
          ],
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "hsl(0, 100%, 20%)",
            "flex": 1,
            "fontSize": 16,
            "paddingHorizontal": 8,
          },
          Object {},
        ]
      }
    >
      Something went wrong
    </Text>
  </View>,
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "hsl(190, 95%, 30%)",
          "borderRadius": 4,
          "justifyContent": "center",
          "paddingHorizontal": 16,
          "paddingVertical": 12,
        },
        undefined,
        Object {
          "marginHorizontal": 16,
          "marginVertical": 12,
        },
      ]
    }
    testID="amplify__button"
  >
    <Text
      style={
        Array [
          Object {
            "color": "hsl(0, 0%, 100%)",
            "fontSize": 16,
            "fontWeight": "700",
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Submit
    </Text>
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 4,
            "justifyContent": "center",
            "paddingHorizontal": 16,
            "paddingVertical": 12,
          },
          undefined,
          Object {
            "marginVertical": 8,
            "minWidth": "50%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "hsl(190, 95%, 30%)",
              "fontSize": 16,
              "fontWeight": "700",
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Skip
      </Text>
    </View>
  </View>,
]
`;
